src/main/java/io/vertx/cassandra/impl/CassandraRowStreamImpl.java

```diff
@@ -0,0 +1,112 @@
+package io.vertx.cassandra.impl;
+
+import com.datastax.driver.core.ResultSet;
+import io.vertx.cassandra.CassandraRowStream;
+import io.vertx.cassandra.Row;
+import io.vertx.core.Context;
+import io.vertx.core.Future;
+import io.vertx.core.Handler;
+import io.vertx.core.Vertx;
+
+import java.util.Iterator;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+
+public class CassandraRowStreamImpl implements CassandraRowStream {
+
+  private final ResultSet datastaxResultSet;
+  private final Vertx vertx;
+  private final Iterator<com.datastax.driver.core.Row> resultSetIterator;
+  private final Lock iteratorAccessLock = new ReentrantLock();
+  private final Context context;
+
+  private volatile Handler<Throwable> exceptionHandler;
```

Comment:
If you use synchronization, `volatile` shouldn't be needed anymore.

Reply:
Done